### PR TITLE
[wasm] Wasm.Build.Tests - use the new sdk with templates

### PR DESCRIPTION
**Pull request by radical**
- based on changes from https://github.com/dotnet/runtime/pull/85311, and https://github.com/dotnet/runtime/pull/85408

### CODE REVIEW

File: src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/Microsoft.NET.Sdk.WebAssembly.Pack.pkgproj [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
   <ItemGroup>
     <ProjectReference Include="$(RepoTasksDir)Microsoft.NET.Sdk.WebAssembly.Pack.Tasks\Microsoft.NET.Sdk.WebAssembly.Pack.Tasks.csproj" />
-    <PackageFile Include="build\*.props;build\*.targets;build\*.web.config" TargetPath="build" />
+    <PackageFile Include="Sdk\*.props;Sdk\*.targets;Sdk\*.web.config" TargetPath="Sdk" />
```
Comment (maraf) [inline src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/Microsoft.NET.Sdk.WebAssembly.Pack.pkgproj:10]: This didn't worked for me. How is it that it works now? Will it still work in blazor without workload?

File: src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Current.Manifest/WorkloadManifest.json.in [hunk 2]
```diff
@@ -178,6 +179,10 @@
       "kind": "Sdk",
       "version": "${PackageVersion}"
     },
+    "Microsoft.NET.Sdk.WebAssembly.Pack": {
```
Comment (maraf) [inline src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Current.Manifest/WorkloadManifest.json.in:182]: What is the consequence of this change? Will it still work in blazor without workload?

File: src/mono/wasm/templates/templates/browser/wwwroot/main.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { dotnet } from './_framework/dotnet.js'
+
+const { setModuleImports, getAssemblyExports, getConfig } = await dotnet
+    .withStartupOptions({})
```
Comment (maraf) [inline src/mono/wasm/templates/templates/browser/wwwroot/main.js:7]: I'm tackling with this my original PR. It shouldn't be in the template

**Review by maraf — COMMENTED** [review]
What is the intended scope of this PR? 
Do you want to take over also my templates PR? 
Do you want to address the JS API needed changes?

File: src/mono/wasm/templates/templates/browser/wwwroot/main.js (hunk 3 above)
Comment (radical) [reply src/mono/wasm/templates/templates/browser/wwwroot/main.js:7]: I can remove this from this draft PR when your changes land.

File: src/mono/wasm/Wasm.Build.Tests/WasmSdk.props [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+<Project ToolsVersion="14.0">
+  <PropertyGroup>
+    <!-- FIXME: Important to get rid of missing workload -->
+    <UsingMicrosoftNETSdkBlazorWebAssembly>true</UsingMicrosoftNETSdkBlazorWebAssembly>
+    <UsingMicrosoftNETSdkWebAssembly>true</UsingMicrosoftNETSdkWebAssembly>
+
+    <!-- WASM projects defaults to browser-wasm -->
+    <RuntimeIdentifier Condition="'$(RuntimeIdentifier)' == ''">browser-wasm</RuntimeIdentifier>
+    <UseMonoRuntime>true</UseMonoRuntime>
+
```
Comment (maraf) [inline src/mono/wasm/Wasm.Build.Tests/WasmSdk.props:21]: This file has a slight change in SDK https://github.com/dotnet/sdk/pull/32170

**Pull request closed without merging**